util/cmdtest/grep.go

```diff
@@ -91,17 +93,127 @@ func (s *IntegrationSuite) Grep(r io.Reader, substr string) (bool, *bytes.Buffer
 	select {
 	case <-time.After(GrepTimeout):
 		if extraDebug {
-			fmt.Printf(" >>>> Grep Timeout reached")
+			fmt.Printf(" >>>> %s Timeout reached", caller)
 		}
 
 		break
 	case found = <-foundch:
 	}
 
 	if extraDebug {
-		fmt.Printf("----------------\nGrep called for substr %q. Found: %v. Read:\n%s\n\n", substr, found, buf.String())
 		fmt.Printf("The complete command output so far:\n%s", s.logBuf.String())
 	}
 
 	return found, buf
 }
+
+func (s *IntegrationSuite) iterAndFind(str []string, fn func(string) bool, caller string) bool {
+	var found bool
+
+	foundch := make(chan bool, 1)
+	go func() {
```

Comment:
I don't see why you need goroutine here. You already have an array of strings. You don't need to wait for anything.

Reply:
You're right I forgot to remove it

Reply:
Addressed [here](https://github.com/src-d/lookout/pull/458/commits/4e8405868224d4b0d5fbf307dd514b67462eee60)